### PR TITLE
Fixed bug in "npm show"

### LINKED ISSUE #4964: [BUG] npm info exits with zero exit code if package@version is not found
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

If I do:

```
npm info npm@8.7.1 && echo $?
```

Then the exit code is 0, but no output from `npm info` is printed, as `8.7.1` of npm doesn't exist. I would expect `npm info` to give a non-zero exit code if the package at the specified version doesn't exist. If the package doesn't exist at all, then `npm info` does have a non-zero exit code (with logs telling you it doesn't exist)

### Expected Behavior

```
npm info npm@8.7.1
```

Would log a message saying the specified version of that package doesn't exist, and exit with a non-zero exit code, much like for non-existent packages.

### Steps To Reproduce

See above, assuming `npm@8.7.1` hasn't been published in the meantime.

### Environment

- npm: 8.7.1
- Node.js: v16.13.1
- OS Name: MacOS latest
- System Model Name: 
- npm config: no non-default configuration.

### PR DESCRIPTION
**Pull request by lukas-kuhn**
This fixes an error in `npm show`. When calling `npm show` with a specific version of a package that does not exist, it does not show anything and gives a zero exit code. This has been changed: now it gives a 404 Error similar to if the package does not exist. Can be tested with `npm show express@5.0.0` (local: `node bin/npm-cli.js info express@5.0.0`)

## References
Fixes #4964

### CODE REVIEW

File: lib/commands/view.js [hunk 1]
```diff
@@ -236,6 +236,15 @@ class View extends BaseCommand {
       }
     })
 
+    //No data has been pushed because no data is matching the specified version
+    if (data.length == 0) {
+      const er = new Error(`No match found for version ${version}`)
+      er.statusCode = 402
```
Comment (ljharb) [inline lib/commands/view.js]: 402 is "payment required"; did you mean 404?
```suggestion
      er.statusCode = 404
```

Comment (lukas-kuhn) [reply lib/commands/view.js]: Yes, thank you!

**Comment by lukas-kuhn** [comment]
Tests are now running properly

File: lib/commands/view.js [hunk 2]
```diff
@@ -236,8 +236,8 @@ class View extends BaseCommand {
       }
     })
 
-    //No data has been pushed because no data is matching the specified version
-    if (data.length == 0) {
+    //  No data has been pushed because no data is matching the specified version
```
Comment (ljharb) [inline lib/commands/view.js]:
```suggestion
    // No data has been pushed because no data is matching the specified version
```

**Comment by wraithgar** [comment]
It looks like some debug logs accidentally got added?

**Comment by lukas-kuhn** [comment]
Forgot to use single quotes. @wraithgar what do you mean exactly?

File: test/lib/commands/tap-testdir-config-completion-sandbox/cache/_logs/2022-06-12T19_00_57_659Z-debug-0.log [hunk 3]
```diff
@@ -0,0 +1,30 @@
+0 timing npm:load:whichnode Completed in 1ms
```
Comment (wraithgar) [inline test/lib/commands/tap-testdir-config-completion-sandbox/cache/_logs/2022-06-12T19_00_57_659Z-debug-0.log]: @lukaskuhn-lku these debug files

File: test/lib/commands/tap-testdir-config-completion-sandbox/cache/_logs/2022-06-12T19_00_57_809Z-debug-0.log [hunk 4]
```diff
@@ -0,0 +1,30 @@
+0 timing npm:load:whichnode Completed in 1ms
```
Comment (wraithgar) [inline test/lib/commands/tap-testdir-config-completion-sandbox/cache/_logs/2022-06-12T19_00_57_809Z-debug-0.log]: @lukaskuhn-lku these debug files

**Comment by lukas-kuhn** [comment]
@wraithgar thanks, removed those files

**Comment by lukaskuhn-lku** [comment]
What is the issue with the failed tests? Coverage?

**Comment by lukas-kuhn** [comment]
Added test for the bug fix so coverage should be back at 100%

**Comment by wraithgar** [comment]
Closing in favor of #5035, which is this PR with one commit added to fix tests. Thanks for making this PR!

**Pull request closed without merging**